packages/salesforcedx-vscode-core/src/commands/commands.ts

```diff
@@ -24,6 +24,70 @@ export interface PreconditionChecker {
   check(): boolean;
 }
 
+export interface PostconditionChecker<T> {
+  check(
+    inputs: ContinueResponse<T> | CancelResponse
+  ): Promise<ContinueResponse<T> | CancelResponse>;
+}
+
+export class FilePathExistsChecker
```

Comment:
This will check if the file path given by the user already contains the file name they specified. If it does, then a warning prompt will show asking to cancel will show. Yes will pass the ContinueResponse along, No will replace with CancelResponse

Reply:
It seems like it would better to create two postcondition checkers: one for the general case and one for the Lightning case. That way, it's easier to understand what is going on instead of having to read this `FilePathExistsChecker` and figure out why there is an 'if' statement case.

Then for the Lightning commands, those will specifically use the Lightning-specific one.